src/PostgREST/Auth/JwtCache.hs

```diff
@@ -20,28 +19,38 @@ import Data.Time.Clock       (UTCTime, nominalDiffTimeToSeconds)
 import Data.Time.Clock.POSIX (utcTimeToPOSIXSeconds)
 import System.Clock          (TimeSpec (..))
 
-import PostgREST.Auth.Types (AuthResult (..))
-import PostgREST.Error      (Error (..))
+import PostgREST.Auth.Types  (AuthResult (..))
+import PostgREST.Error       (Error (..))
+import PostgREST.Internal    (recursiveSizeNF)
+import PostgREST.Observation (Observation (..), ObservationHandler)
 
+import Control.Debounce
 import Protolude
 
-newtype JwtCacheState = JwtCacheState
-  { jwtCache :: C.Cache ByteString AuthResult
+type SizeInBytes = Int
+
+-- See: https://github.com/PostgREST/postgrest/pull/3802#discussion_r1971074445
+-- to understand why we are also storing the size in the cache
```

Comment:
Best to inline this explanation and be as specific as possible. Also mentioning why would it work when purging.